sample.bluetooth.peripheral_hr_rv32m1_vega_ri5cy does not build

Showed up in https://buildkite.com/zephyr/zephyr-daily/builds/229

Local build shows additional failures:
```
[97/204] Building C object zephyr/subsys/bluetooth/controller/CMakeFiles/subsys__bluetooth__controller.dir/util/memq.c.obj
/mnt/nordic/zp/zephyr/subsys/bluetooth/controller/util/memq.c: In function 'memq_enqueue':
/mnt/nordic/zp/zephyr/subsys/bluetooth/controller/util/memq.c:103:2: warning: implicit declaration of function 'cpu_dmb' [-Wimplicit-function-declaration]
  103 |  cpu_dmb(); /* Ensure data accesses are synchronized */
      |  ^~~~~~~
[121/204] Building C object zephyr/subsys/bluetooth/controller/CMakeFiles/subsys__bluetooth__controller.dir/ll_sw/openisa/hal/RV32M1/radio/radio.c.obj
FAILED: zephyr/subsys/bluetooth/controller/CMakeFiles/subsys__bluetooth__controller.dir/ll_sw/openisa/hal/RV32M1/radio/radio.c.obj 
/usr/local/zephyr-sdk/riscv64-zephyr-elf/bin/riscv64-zephyr-elf-gcc -DBUILD_VERSION=v2.5.0-rc2-5-g5b8d79fe9a16 -DCPU_RV32M1_ri5cy -DKERNEL -DXCVR_GENFSK_ENABLED -D_FORTIFY_SOURCE=2 -D__ZEPHYR__=1 -I../../../../subsys/bluetooth/controller/ll_sw/openisa/lll -I../../../../subsys/bluetooth/controller/ll_sw/openisa -I../../../../subsys/bluetooth/controller/hci/openisa -I../../../../subsys/bluetooth/controller/. -I../../../../subsys/bluetooth/controller/include -I../../../../subsys/bluetooth/controller/ll_sw -I../../../../include -Izephyr/include/generated -I../../../../soc/riscv/openisa_rv32m1 -I../../../../subsys/bluetooth -I/mnt/nordic/zp/modules/hal/openisa/vega_sdk_riscv/RISCV -I/mnt/nordic/zp/modules/hal/openisa/vega_sdk_riscv/devices/RV32M1 -I/mnt/nordic/zp/modules/hal/openisa/vega_sdk_riscv/devices/RV32M1/drivers/. -I/mnt/nordic/zp/modules/hal/openisa/vega_sdk_riscv/middleware/wireless/framework/XCVR/RV32M1 -I/mnt/nordic/zp/modules/crypto/tinycrypt/lib/include -isystem ../../../../lib/libc/minimal/include -isystem /usr/local/zephyr-sdk-0.12.2/riscv64-zephyr-elf/bin/../lib/gcc/riscv64-zephyr-elf/10.2.0/include -isystem /usr/local/zephyr-sdk-0.12.2/riscv64-zephyr-elf/bin/../lib/gcc/riscv64-zephyr-elf/10.2.0/include-fixed -Os -imacros /mnt/nordic/zp/zephyr/samples/bluetooth/peripheral_hr/build/zephyr/include/generated/autoconf.h -ffreestanding -fno-common -g -mabi=ilp32 -march=rv32ima -imacros /mnt/nordic/zp/zephyr/include/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-main -Wno-pointer-sign -Wpointer-arith -Wno-address-of-packed-member -Wno-unused-but-set-variable -Werror=implicit-int -fno-asynchronous-unwind-tables -fno-pie -fno-pic -fno-strict-overflow -fno-reorder-functions -fno-defer-pop -fmacro-prefix-map=/mnt/nordic/zp/zephyr/samples/bluetooth/peripheral_hr=CMAKE_SOURCE_DIR -fmacro-prefix-map=/mnt/nordic/zp/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/mnt/nordic/zp=WEST_TOPDIR -ffunction-sections -fdata-sections -march=rv32imc -std=c99 -nostdinc -O2 -MD -MT zephyr/subsys/bluetooth/controller/CMakeFiles/subsys__bluetooth__controller.dir/ll_sw/openisa/hal/RV32M1/radio/radio.c.obj -MF zephyr/subsys/bluetooth/controller/CMakeFiles/subsys__bluetooth__controller.dir/ll_sw/openisa/hal/RV32M1/radio/radio.c.obj.d -o zephyr/subsys/bluetooth/controller/CMakeFiles/subsys__bluetooth__controller.dir/ll_sw/openisa/hal/RV32M1/radio/radio.c.obj   -c /mnt/nordic/zp/zephyr/subsys/bluetooth/controller/ll_sw/openisa/hal/RV32M1/radio/radio.c
/mnt/nordic/zp/zephyr/subsys/bluetooth/controller/ll_sw/openisa/hal/RV32M1/radio/radio.c: In function 'radio_sleep':
/mnt/nordic/zp/zephyr/subsys/bluetooth/controller/ll_sw/openisa/hal/RV32M1/radio/radio.c:1535:2: error: #error "Missing atomic operation in radio_sleep()"
 1535 | #error "Missing atomic operation in radio_sleep()"
      |  ^~~~~
/mnt/nordic/zp/zephyr/subsys/bluetooth/controller/ll_sw/openisa/hal/RV32M1/radio/radio.c: In function 'radio_wake':
/mnt/nordic/zp/zephyr/subsys/bluetooth/controller/ll_sw/openisa/hal/RV32M1/radio/radio.c:1584:2: error: #error "Missing atomic operation in radio_wake()"
 1584 | #error "Missing atomic operation in radio_wake()"
      |  ^~~~~
[135/204] Building C object zephyr/subsys/bluetooth/controller/CMakeFiles/subsys__bluetooth__controller.dir/ll_sw/openisa/lll/lll_conn.c.obj
In file included from /mnt/nordic/zp/zephyr/subsys/bluetooth/controller/ll_sw/openisa/lll/lll_conn.c:16:
../../../../subsys/bluetooth/controller/./util/mfifo.h: In function 'mfifo_by_idx_enqueue':
../../../../subsys/bluetooth/controller/./util/mfifo.h:126:2: warning: implicit declaration of function 'cpu_dmb' [-Wimplicit-function-declaration]
  126 |  cpu_dmb(); /* Ensure data accesses are synchronized */
      |  ^~~~~~~
[141/204] Building C object zephyr/subsys/bluetooth/controller/CMakeFiles/subsys__bluetooth__controller.dir/ll_sw/openisa/lll/lll_adv.c.obj
In file included from /mnt/nordic/zp/zephyr/subsys/bluetooth/controller/ll_sw/openisa/lll/lll_adv.c:23:
../../../../subsys/bluetooth/controller/./util/mfifo.h: In function 'mfifo_by_idx_enqueue':
../../../../subsys/bluetooth/controller/./util/mfifo.h:126:2: warning: implicit declaration of function 'cpu_dmb' [-Wimplicit-function-declaration]
  126 |  cpu_dmb(); /* Ensure data accesses are synchronized */
      |  ^~~~~~~
[144/204] Building C object zephyr/subsys/bluetooth/controller/CMakeFiles/subsys__bluetooth__controller.dir/ll_sw/ull.c.obj
In file included from /mnt/nordic/zp/zephyr/subsys/bluetooth/controller/ll_sw/ull.c:24:
../../../../subsys/bluetooth/controller/./util/mfifo.h: In function 'mfifo_by_idx_enqueue':
../../../../subsys/bluetooth/controller/./util/mfifo.h:126:2: warning: implicit declaration of function 'cpu_dmb' [-Wimplicit-function-declaration]
  126 |  cpu_dmb(); /* Ensure data accesses are synchronized */
      |  ^~~~~~~
[153/204] Building C object zephyr/subsys/bluetooth/controller/CMakeFiles/subsys__bluetooth__controller.dir/ll_sw/ull_conn.c.obj
In file included from /mnt/nordic/zp/zephyr/subsys/bluetooth/controller/ll_sw/ull_conn.c:22:
../../../../subsys/bluetooth/controller/./util/mfifo.h: In function 'mfifo_by_idx_enqueue':
../../../../subsys/bluetooth/controller/./util/mfifo.h:126:2: warning: implicit declaration of function 'cpu_dmb' [-Wimplicit-function-declaration]
  126 |  cpu_dmb(); /* Ensure data accesses are synchronized */
      |  ^~~~~~~
[154/204] Building C object zephyr/subsys/bluetooth/host/CMakeFiles/subsys__bluetooth__host.dir/hci_core.c.obj
ninja: build stopped: subcommand failed.
```

--- a/subsys/bluetooth/controller/Kconfig.ll_sw_split
+++ b/subsys/bluetooth/controller/Kconfig.ll_sw_split
@@ -40,6 +40,7 @@ config BT_LLL_VENDOR_NORDIC
 	select BT_CTLR_XTAL_ADVANCED_SUPPORT
 	select BT_CTLR_SCHED_ADVANCED_SUPPORT
 	select BT_CTLR_TIFS_HW_SUPPORT
+	select BT_CTLR_ULL_LLL_PRIO_SUPPORT
 
 	default y
 	help
@@ -64,6 +65,9 @@ config BT_CTLR_SCHED_ADVANCED_SUPPORT
 	bool
 
 config BT_CTLR_TIFS_HW_SUPPORT
+	bool
+
+config BT_CTLR_ULL_LLL_PRIO_SUPPORT
 	bool
 
 config BT_CTLR_RX_PRIO_STACK_SIZE
@@ -274,7 +278,7 @@ config BT_CTLR_SCHED_ADVANCED
 	  leading to skipped events amongst active roles.
 
 config BT_CTLR_LLL_PRIO
-	int "Lower Link Layer (Radio) IRQ priority" if !BT_CTLR_ZLI
+	int "Lower Link Layer (Radio) IRQ priority" if (BT_CTLR_ULL_LLL_PRIO_SUPPORT && !BT_CTLR_ZLI)
 	range 0 3 if SOC_SERIES_NRF51X
 	range 0 6 if (SOC_SERIES_NRF52X || SOC_SERIES_NRF53X)
 	default 0
@@ -282,17 +286,17 @@ config BT_CTLR_LLL_PRIO
 	  The interrupt priority for event preparation and radio IRQ.
 
 config BT_CTLR_ULL_HIGH_PRIO
-	int "Upper Link Layer High IRQ priority"
+	int "Upper Link Layer High IRQ priority" if BT_CTLR_ULL_LLL_PRIO_SUPPORT
 	range BT_CTLR_LLL_PRIO 3 if SOC_SERIES_NRF51X
 	range BT_CTLR_LLL_PRIO 6 if (SOC_SERIES_NRF52X || SOC_SERIES_NRF53X)
-	default BT_CTLR_LLL_PRIO if (BT_CTLR_ZLI || BT_CTLR_LOW_LAT)
+	default BT_CTLR_LLL_PRIO if (!BT_CTLR_ULL_LLL_PRIO_SUPPORT || BT_CTLR_ZLI || BT_CTLR_LOW_LAT)
 	default 1
 	help
 	  The interrupt priority for Ticker's Worker IRQ and Upper Link Layer
 	  higher priority functions.
 
 config BT_CTLR_ULL_LOW_PRIO
-	int "Upper Link Layer Low IRQ priority"
+	int "Upper Link Layer Low IRQ priority" if BT_CTLR_ULL_LLL_PRIO_SUPPORT
 	range BT_CTLR_ULL_HIGH_PRIO 3 if SOC_SERIES_NRF51X
 	range BT_CTLR_ULL_HIGH_PRIO 6 if (SOC_SERIES_NRF52X || SOC_SERIES_NRF53X)
 	default BT_CTLR_ULL_HIGH_PRIO

--- a/subsys/bluetooth/controller/Kconfig.ll_sw_split
+++ b/subsys/bluetooth/controller/Kconfig.ll_sw_split
@@ -285,7 +285,7 @@ config BT_CTLR_ULL_HIGH_PRIO
 	int "Upper Link Layer High IRQ priority"
 	range BT_CTLR_LLL_PRIO 3 if SOC_SERIES_NRF51X
 	range BT_CTLR_LLL_PRIO 6 if (SOC_SERIES_NRF52X || SOC_SERIES_NRF53X)
-	default BT_CTLR_LLL_PRIO if (BT_CTLR_ZLI || BT_CTLR_LOW_LAT_ULL)
+	default BT_CTLR_LLL_PRIO if (BT_CTLR_ZLI || BT_CTLR_LOW_LAT)
 	default 1
 	help
 	  The interrupt priority for Ticker's Worker IRQ and Upper Link Layer

--- a/subsys/bluetooth/controller/ll_sw/openisa/hal/cpu.h
+++ b/subsys/bluetooth/controller/ll_sw/openisa/hal/cpu.h
@@ -11,3 +11,12 @@ static inline void cpu_sleep(void)
 	/* __SEV(); */
 	__WFE();
 }
+
+static inline void cpu_dmb(void)
+{
+	/* FIXME: Add necessary host machine required Data Memory Barrier
+	 *        instruction alongwith the below defined compiler memory
+	 *        clobber.
+	 */
+	__asm__ volatile ("" : : : "memory");
+}

--- a/subsys/bluetooth/controller/ll_sw/openisa/lll/lll_adv.c
+++ b/subsys/bluetooth/controller/ll_sw/openisa/lll/lll_adv.c
@@ -7,12 +7,12 @@
 #include <stdbool.h>
 #include <stddef.h>
 
-#include <zephyr/types.h>
+#include <toolchain.h>
+#include <soc.h>
+#include <bluetooth/hci.h>
 #include <sys/byteorder.h>
 
-#include <toolchain.h>
-#include <bluetooth/hci.h>
-
+#include "hal/cpu.h"
 #include "hal/ccm.h"
 #include "hal/radio.h"
 #include "hal/ticker.h"
@@ -40,7 +40,6 @@
 
 #define LOG_MODULE_NAME bt_ctlr_llsw_openisa_lll_adv
 #include "common/log.h"
-#include <soc.h>
 #include "hal/debug.h"
 
 static int init_reset(void);
@@ -212,14 +211,11 @@ struct pdu_adv *lll_adv_pdu_alloc(struct lll_adv_pdu *pdu, uint8_t *idx)
 		uint8_t first_latest;
 
 		pdu->last = first;
-		/* FIXME: Ensure that data is synchronized so that an ISR
+		/* NOTE: Ensure that data is synchronized so that an ISR
 		 *        vectored, after pdu->last has been updated, does
-		 *        access the latest value. __DMB() is used in ARM
-		 *        Cortex M4 architectures. Use appropriate
-		 *        instructions on other platforms.
-		 *
-		 *        cpu_dmb();
+		 *        access the latest value.
 		 */
+		cpu_dmb();
 		first_latest = pdu->first;
 		if (first_latest != first) {
 			last++;

--- a/subsys/bluetooth/controller/ll_sw/openisa/lll/lll_conn.c
+++ b/subsys/bluetooth/controller/ll_sw/openisa/lll/lll_conn.c
@@ -8,15 +8,16 @@
 #include <stddef.h>
 
 #include <toolchain.h>
-#include <zephyr/types.h>
+#include <soc.h>
 #include <sys/util.h>
+
+#include "hal/cpu.h"
+#include "hal/ccm.h"
+#include "hal/radio.h"
 
 #include "util/mem.h"
 #include "util/memq.h"
 #include "util/mfifo.h"
-
-#include "hal/ccm.h"
-#include "hal/radio.h"
 
 #include "pdu.h"
 
@@ -29,7 +30,6 @@
 
 #define LOG_MODULE_NAME bt_ctlr_llsw_openisa_lll_conn
 #include "common/log.h"
-#include <soc.h>
 #include "hal/debug.h"
 
 static int init_reset(void);

